### DISCUSSION TITLE
Create app with smallrye-reactive-messaging-kafka and Kotlin

### DISCUSSION THREAD

**Discussion by nmodinMerkle** [quarkusio/quarkus#33153] [Q&A] [answered]
I get some problems when running the following to create an app using Kafka Messaging and Kotlin:

`quarkus create app com.leara:leara-api --gradle-kotlin-dsl --extension='kotlin,resteasy-reactive-jackson,smallrye-reactive-messaging-kafka'
`
I get the starter code for the rest resource (GreetingResource), but not messaging application (MyReactiveMessagingApplication). When running the example things work as expected and I get the
endpoint provided GreetingResource up and running.

When doing the same for Java, I get the starter code for both resteasy and kafka, and the example work OOTB.
At startup, there are a few messages sent over a couple of different channels.

If I go ahead and copy MyReactiveMessagingApplication and convert it to Kotlin and include it in the Kotlin project, I see other issue when running ./gradlwe devQuarkus.

These issues seems to come from Kafka and the container starts, perhaps in conjunction with some config issues:

`WARN  [org.apa.kaf.cli.pro.int.Sender] (kafka-producer-network-thread | kafka-producer-words-out) [Producer clientId=kafka-producer-words-out] Received invalid metadata error in produce request on partition words-out-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition... Going to request metadata update now`

My question I guess is what's wrong in the Kotlin case? :) I don't mind copying / creating my Kafka messaging code myself, but I can't seem to find what is wrong with the Kafka errors I see in the log.

[a comment by an automated account is not shown]

**Comment by ozangunalp** [comment]
I don't think the error you've cited is relevant. It is a retriable exception from Kafka producer and the send operation will be retried. It means that the producer couldn't get the partition metadata from the broker. 

I'll try it to see what's going on for that example in Kotlin.

**Comment by ozangunalp** [comment, marked as answer]
@nmodinMerkle I just created a project with kotlin and reactive messaging kafka and tested this case. Everything seems to work well. 
I think the retriable exception you are seeing is caused by the redpanda image we are using as dev service. It happens when the topic is auto-created on produced record.

You can change the dev service provider using `quarkus.kafka.devservices.provider` property. You can choose between `redpanda` (default), `kafka-native`, or `strimzi`.
